File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/lists/Team.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */
using System;
using System.Collections;
using com.db4odoc.f1.evaluations;

namespace com.db4odoc.f1.lists
{
	public class Team
	{
		private IList _pilots;
		private string _name;

		public Team()
		{
			_pilots = CollectionFactory.newList();	
		}

		public string Name
		{
			get
			{
				return _name;
			}
            
			set
			{
				_name = value;
			}
		}

		public void AddPilot(Pilot pilot)
		{
			_pilots.Add(pilot);
		}
	
		public Pilot GetPilot(int index)
		{
			return (Pilot)_pilots[index]; 
		}
	
		public void RemovePilot(int index)
		{
			_pilots.Remove(index);
		}
	
		public void UpdatePilot(int index, Pilot newPilot)
		{
			_pilots[index] = newPilot;
		}
	}
}

File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/lists/Team.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 

Imports System
Imports System.Collections
Imports com.db4odoc.f1.evaluations

Namespace com.db4odoc.f1.lists
    Public Class Team
        Private _pilots As IList
        Private _name As String

        Public Sub New()
            _pilots = CollectionFactory.NewList()
        End Sub

        Public Property Name() As String
            Get
                Return _name
            End Get
            Set(ByVal Value As String)
                _name = Value
            End Set
        End Property

        Public Sub AddPilot(ByVal pilot As evaluations.Pilot)
            _pilots.Add(pilot)
        End Sub

        Public Function GetPilot(ByVal index As Integer) As evaluations.Pilot
            Return CType(_pilots(index), evaluations.Pilot)
        End Function

        Public Sub RemovePilot(ByVal index As Integer)
            _pilots.Remove(index)
        End Sub

        Public Sub UpdatePilot(ByVal index As Integer, ByVal NewPilot As evaluations.Pilot)
            _pilots(index) = NewPilot
        End Sub
    End Class
End Namespace
